collector/collector.go

```diff
@@ -154,6 +170,32 @@ func (c *Collector) GetRawNetworkMetricsHandler(w http.ResponseWriter, req *http
 	}
 }
 
+func (c *Collector) writeGzipBody(w http.ResponseWriter, body []byte) error {
+	writer, err := gzip.NewWriterLevel(w, gzip.BestCompression)
+	if err != nil {
```

Comment:
NIT: It's probably not worth checking the error in this case.

> The error returned will be nil if the level is valid. 

https://pkg.go.dev/compress/gzip#NewWriterLevel